cmake/platform/windows.cmake

```diff
@@ -88,3 +88,41 @@ if(MSVC)
   # https://blogs.msdn.microsoft.com/vcblog/2018/04/09/msvc-now-correctly-reports-__cplusplus/
   add_compile_options(/Zc:__cplusplus)
 endif()
+
+#
+# Helper function to add Python executable wrapper on Windows due to lack of shebang support
+#
+# Code courtesy of Yujin Robot.  Derived from https://github.com/ros-windows/win_ros
+#
+
+#
+# Add Python executable wrapper around Python scripts on Windows.
+# Python scripts with (or without) .py extension are not executable on Windows due to lack of shebang support
+#
+# :param script: Python script name that needs a wrapper
+# :type script: string
+# :param target: build target name
+# :type target: string
+#
+# @public
+#
+function(add_python_executable_helper)
+  set(options)
+  set(oneValueArgs SCRIPT_NAME TARGET_NAME DESTINATION)
+  set(multiValueArgs)
+  cmake_parse_arguments(add_python_executable_helper "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+  if (WIN32)
+    if (NOT TARGET ${add_python_executable_helper_TARGET_NAME})
```

Comment:
Why should this silently skip the executable creation and installation if the target already exists?

Reply:
a few reasons for this:
- same function might get called multiple times for Python scripts of the same name (e.g., for scripts listed in `setup.py`, `add_python_executable` will be called twice in `catkin_python_setup` to install into both `devel\bin` and `install\bin`; also, there have been cases where a Python script appears in `catkin_install_python` while it is included in `setup.py`, so win32 wrapper will need to be created for both of those occasions)
- because of the above reason, without `if (NOT TARGET ...)`, `add_executable` would hit this error
```
CMake Error at cmake/platform/windows.cmake:119 (add_executable):
  add_executable cannot create target "catkin_find_executable" because
  another target with the same name already exists.  The existing target is
  an executable created in source directory
  "C:/ros/catkin_ws/upstream/catkin/src/catkin".  See documentation for
  policy CMP0002 for more details.
Call Stack (most recent call first):
  cmake/catkin_python_setup.cmake:160 (add_python_executable)
  CMakeLists.txt:23 (catkin_python_setup)
```
- also this could probably same time during compilation as executables of the same name should do the exact same thing (based on current implementation)

Reply:
> `add_python_executable` will be called twice in `catkin_python_setup` to install into both `devel\bin` and `install\bin`

I would argue that both calls should pass different target names in that case.

Reply:
> I would argue that both calls should pass different target names in that case.

change to remove `if (NOT TARGET ...)` mask